share/revng/test/tests/abi/runner.c

```diff
@@ -0,0 +1,226 @@
+/*
+ * This file is distributed under the MIT License. See LICENSE.md for details.
+ */
+
+/* RUN-default: nope */
+
+#include <assert.h>
+#include <errno.h>
+#include <fcntl.h>
+#include <setjmp.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "gathered_symbols.h"
+#include "ucontext_wrapper.h"
+
```

Comment:
Add a comment briefly explaining the idea of this program. Like: we have some metadata from a `.h` file, we `mmap` a file, we used the metadata to introduce breakpoints here and there, dump the state in the signal handler and restore the original instruction, resume execution, etc.

Putting this documentation somewhere else just increases the risk of it gettting outdated.

Reply:
Looks great!